packages/event-handler/tests/unit/rest/BaseRouter.test.ts

```diff
@@ -179,23 +226,14 @@ describe('Class: BaseRouter', () => {
     expect(logger.debug).toHaveBeenCalledWith('test debug');
   });
 
-  describe('middleware', () => {
+  describe('middleware - global', () => {
```

Comment:
I think if possible, I'd remove the usage of nested `describe` blocks in favor of `// #region middleware` & `// endregion`  [comments](https://stackoverflow.com/a/76568099), so we avoid the nesting but we still get the folding. Same for the other route specific middleware group.

This will also remove the SonarCloud issue.

Reply:
As mentioned offline, I will break this test suite into multiple files in a separate PR as it's getting too big.